azure/scope/cluster.go

```diff
@@ -87,6 +88,10 @@ func NewClusterScope(ctx context.Context, params ClusterScopeParams) (*ClusterSc
 		}
 	}
 
+	if params.Cache == nil {
```

Comment:
since Cache isn't strictly required, why fail if it's nil? we could simply initialize it to an empty cache if nil (that way no need to add the empty value everywhere in the controllers)

Reply:
This will work, thanks!